projects/centre_console/src/power_fsm.c

```diff
@@ -1,86 +1,75 @@
   FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_FAULT, power_state_fault);
 }
 
 FSM_STATE_TRANSITION(power_state_main) {
-  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_OFF, power_state_turning_off);
+  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_OFF, power_state_transitioning);
   FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_FAULT, power_state_fault);
 }
 
 FSM_STATE_TRANSITION(power_state_aux) {
-  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_ON_MAIN, power_state_turning_on_main);
-  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_OFF, power_state_turning_off);
+  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_ON_MAIN, power_state_transitioning);
+  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_OFF, power_state_transitioning);
   FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_FAULT, power_state_fault);
 }
 
-FSM_STATE_TRANSITION(power_state_turning_on_main) {
+FSM_STATE_TRANSITION(power_state_transitioning) {
   FSM_ADD_TRANSITION(POWER_MAIN_SEQUENCE_EVENT_COMPLETE, power_state_main);
-  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_FAULT, power_state_fault);
-}
-
-FSM_STATE_TRANSITION(power_state_turning_on_aux) {
   FSM_ADD_TRANSITION(POWER_AUX_SEQUENCE_EVENT_COMPLETE, power_state_aux);
-  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_FAULT, power_state_fault);
-}
-
-FSM_STATE_TRANSITION(power_state_turning_off) {
   FSM_ADD_TRANSITION(POWER_OFF_SEQUENCE_EVENT_COMPLETE, power_state_off);
   FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_FAULT, power_state_fault);
 }
 
-static bool prv_guard_faults_cleared(const Fsm *fsm, const Event *e, void *context) {
+static bool prv_guard_clear_fault(const Fsm *fsm, const Event *e, void *context) {
   PowerFsmStorage *power_fsm = (PowerFsmStorage *)context;
-  power_fsm->fault_bitset &= ~(e->data);
-  bool faults_cleared = (power_fsm->fault_bitset == POWER_FSM_NO_FAULT);
-  if (!faults_cleared) {
-    event_raise(CENTRE_CONSOLE_POWER_EVENT_PUBLISH_FAULT, power_fsm->fault_bitset);
-  }
-  return faults_cleared;
+  return power_fsm->previous_state == e->data;
 }
 
 FSM_STATE_TRANSITION(power_state_fault) {
-  FSM_ADD_GUARDED_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_CLEAR_FAULT, prv_guard_faults_cleared,
-                             power_state_turning_on_aux);
-  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_FAULT, power_state_fault);
+  FSM_ADD_GUARDED_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_CLEAR_FAULT, prv_guard_clear_fault,
+                             power_state_off);
+  FSM_ADD_GUARDED_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_CLEAR_FAULT, prv_guard_clear_fault,
+                             power_state_aux);
+  FSM_ADD_GUARDED_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_CLEAR_FAULT, prv_guard_clear_fault,
+                             power_state_main);
+  FSM_ADD_TRANSITION(CENTRE_CONSOLE_POWER_EVENT_OFF, power_state_transitioning);
 }
 
 void prv_set_current_state(void *context, PowerState state) {
   PowerFsmStorage *storage = (PowerFsmStorage *)context;
   storage->current_state = state;
 }
```

Comment:
I assume this might've made sense on the old code ... but it's literally one line so I'd say export this logic?

This function does seem useful if it stored the previous state (maybe even the destination state) while setting the current one imo, but you'd have to change your logic accordingly